### PR TITLE
[SDTEST-482] Add get_time_provider setting to avoid timecop mocking the clock_gettime

### PR DESCRIPTION
**Pull request by anmarchenko**
**What does this PR do?**
Adds `get_time_provider` setting similar to `time_now_provider` that provides a workaround for the situation when timecop mocks Process.clock_gettime.

**Motivation:**
Timecop v0.9.9 allows to [mock monotonic clock](https://github.com/travisjeffery/timecop/pull/419), which might affect tests tracing. See the example of this failure [here](https://github.com/DataDog/datadog-ci-rb/pull/242).

**How to test the change?**
Unit tests are provided

### CODE REVIEW

**Comment by TonyCTHsu** [comment]
Is this an internal usage from CI gem? Perhaps we should avoid making it public

**Comment by anmarchenko** [comment]
@TonyCTHsu the option to set Time.now provider is public and documented for customers here: 
https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#setting-the-time-provider

In this case, I see no reason to hide this option from public usage: there are customers that used `time_now_provider`, so they could use `get_time_provider` as well

**Comment by ivoanjo** [comment]
`time_now_provider` was added in https://github.com/DataDog/dd-trace-rb/pull/1224 . I guess the intent was to allow better control over span start/end times?

File: lib/datadog/core/configuration/settings.rb [hunk 1]
```diff
@@ -654,6 +654,33 @@ def initialize(*_)
           end
         end
 
+        # The monotonic clock time provider used by Datadog.
```
Comment (TonyCTHsu) [inline lib/datadog/core/configuration/settings.rb:657]: I think it is worth adding a comment to mention that `datadog-ci` is using this option to generate traces, not skewed by `Timecop` and we consider this internal (not public).

**Review by TonyCTHsu — APPROVED** [review]

**Review by delner — COMMENTED** [review]
I like the idea of configuring in a cooperative way over forcing a mock. My question is more about the name.

`time_provider` suggests that all time in the library will be provided through this provider? Do we have a means of really enforcing this consistent use? If we use `Time.now` somewhere instead of this provider, we may end up with broken expectations around behavior.

Off the top of my head, I feel like this could be either:

1. Commit to using this more broadly
2. OR rename this to `trace_time_provider` and more narrowly bound its use. But then it doesn't belong in `Core`.

WDYT? (I'm not quite sure what to do about this.)

**Comment by anmarchenko** [comment]
@delner I don't have any preference on this topic, so I think whatever works for tracing/core is fine for me

**Review by p-datadog — APPROVED** [review]
Diff LGTM, I have no opinion on the naming/scope.

**Comment by anmarchenko** [comment]
@delner I suggest let's merge it now because it solves a problem for the users, I will leave the question of whether it belongs to Core or Tracing open for now

**Pull request merged**